### ISSUE TITLE
livecd-iso-to-disk error: "The partition name must be 'EFI System Partition'."

### ISSUE THREAD

**Issue by Conan-Kudo** [livecd-tools/livecd-tools#56]
From Red Hat Bugzilla downstream bugs ([rhbz#1437601](https://bugzilla.redhat.com/1437601) and [rhbz#1438578](https://bugzilla.redhat.com/1438578)):

### Description of problem:
        ALERT:
        The partition name must be 'EFI System Partition'.

### Version-Release number of selected component (if applicable):
livecd-tools-24.2-1.fc26.x86_64

### How reproducible:
Always

### Steps to Reproduce:
1. `/usr/bin/livecd-iso-to-disk --efi --format --noverify --reset-mbr ~jamundson/Downloads/Fedora-LXQt-Live-x86_64-26-20170328.n.0.iso /dev/sdc1`

```
Disk /dev/sdc: 14.7 GiB, 15812526080 bytes, 30883840 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: 906DE484-2584-412F-9F3A-0C7919CA2B3F

Device     Start      End  Sectors  Size Type
/dev/sdc1   8192 30879743 30871552 14.7G EFI System
```

### Actual results:
```
/usr/bin/livecd-iso-to-disk: line 492: : command not found

    WARNING: This will DESTROY All DATA on: /dev/sdc !!

        Press Enter to continue, or Ctrl C to abort.

/dev/sdc: 8 bytes were erased at offset 0x00000200 (gpt): 45 46 49 20 50 41 52 54
/dev/sdc: 8 bytes were erased at offset 0x3ae7ffe00 (gpt): 45 46 49 20 50 41 52 54
/dev/sdc: 2 bytes were erased at offset 0x000001fe (PMBR): 55 aa
/dev/sdc: calling ioctl to re-read partition table: Success
Waiting for devices to settle...
mkfs.fat 4.1 (2017-01-24)

        ALERT:
        The partition name must be 'EFI System Partition'.

        This can be set with a partition editor, such as parted,
        or you can run livecd-iso-to-disk with the --reset-mbr option.
```
### Expected results:
Correctly written USB drive.

### Additional info:
Also happened on Fedora 25 before system-upgrade.

A quick glance tells me line 408, or something before it, is wrong -
`    p=${p##$device}`
sets `p` and `partnum` to empty strings.

Confirmed that livecd-tools-24.1-1.fc26.x86_64 does not suffer from this problem.

**Comment by Conan-Kudo** [comment]
Based on the timeline and the information provided in the downstream bugs, it looks like this change may be responsible:

```diff
diff --git a/tools/livecd-iso-to-disk.sh b/tools/livecd-iso-to-disk.sh
index e5f8c0d..47b1229 100755
--- a/tools/livecd-iso-to-disk.sh
+++ b/tools/livecd-iso-to-disk.sh
@@ -361,29 +383,30 @@ getdisk() {
     fi
 
     p=$(udevadm info -q path -n $DEV)
-    if [ $? -gt 0 ]; then
+    if [[ $? -gt 0 ]]; then
         echo "Error getting udev path to $DEV"
         exitclean
     fi
     if [[ -n $loop ]]; then
         node=${DEV#/dev/loop}
-        p=$(basename $DEV)
+        p=${DEV##*/}
         device=loop${node%p*}
-    elif [ -e /sys/$p/device ]; then
-        device=$(basename /sys/$p)
+    elif [[ -e /sys/$p/device ]]; then
+        device=${p##*/}
     else
-        device=$(basename $(readlink -f /sys/$p/../))
+        p=$(readlink -f /sys/$p/../)
+        device=${p##*/}
     fi
-    if [ -z "$device" -o ! -e /sys/block/$device -o ! -e /dev/$device ]; then
+    if [[ -z $device || ! -e /sys/block/$device || ! -e /dev/$device ]]; then
         echo "Error finding block device of $DEV.  Aborting!"
         exitclean
     fi
 
     device="/dev/$device"
     # FIXME: weird dev names could mess this up I guess
-    p=/dev/$(basename $p)
+    p=/dev/${p##*/}
     p=${p##$device}
-    # Strip off leading p from partnum, eg. with /dev/mmcblk0p1
+    # Strip off leading p from partnum, e.g., with /dev/mmcblk0p1
     partnum=${p##p}
 }

```

@FGrose Since you've done quite a bit of hacking on this, do you have any idea what caused this issue?

**Comment by Conan-Kudo** [comment]
This should be fixed by 46ab8bb3cb0b8f86c605d6a26f466f4a8f2a5e00.
